src/coreclr/nativeaot/System.Private.CoreLib/src/System/Array.NativeAot.cs

```diff
@@ -1196,6 +1199,68 @@ public void RemoveAt(int index)
         }
     }
 
+    internal class SZGenericArrayEnumeratorBase : IDisposable
+    {
+        protected int _index;
+        protected int _endIndex;
+
+        internal SZGenericArrayEnumeratorBase()
```

Comment:
would it be better to capture the _endIndex in the construction as the whole point of this base class is to inline and read-only the the length?